tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -84,25 +114,108 @@ def __init__(self, references):
 
     @classmethod
     def from_yaml(cls, loader, node):
-        return UserList(loader.construct_sequence(node))
+        return YamlReferenceTagList(loader.construct_sequence(node))
 
     @classmethod
     def to_yaml(cls, dumper, data):
-        return dumper.represent_sequence(cls.yaml_tag, data.data, flow_style=True)
+        return dumper.represent_sequence(cls.yaml_tag, data, flow_style=True)
+
+
+def convert_to_config_node(node):
+    """
+    Convert json like data to ConfigNode
+    """
+    if isinstance(node, dict):
+        return ConfigNodeDict({k: convert_to_config_node(v) for k, v in node.items()})
+    elif isinstance(node, list):
+        constructor = YamlReferenceTagList if isinstance(node, YamlReferenceTagList) else ConfigNodeList
+
+        return constructor([convert_to_config_node(v) for v in node])
+    else:
+        return node
 
 
-def generate_gitlab_full_configuration(input_file, context=None, compare_to=None):
+def apply_yaml_extends(config: dict, node):
+    """
+    Applies `extends` yaml tags to the node and its children inplace
+    """
+    if not isinstance(node, dict):
+        return
+
+    if 'extends' in node:
+        parents = node['extends']
+        if isinstance(parents, str):
+            parents = [parents]
+
+        # Merge parent
+        for parent_name in parents:
+            parent = config[parent_name]
+            apply_yaml_postprocessing(config, parent)
+            for key, value in parent.items():
+                if key not in node:
+                    node[key] = value
+
+        del node['extends']
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Same here, a comment would be nice. I'm thinking giving a config example. While reviewing the PR I'm helping myself with the config as it makes the code way easier to understand. So having a commented small configuration would be nice for ppl without context !